app/cu_editor.py

```diff
@@ -397,6 +397,45 @@ def setTextBuffer(self, textBuffer):
     })
     self.commandSet = commandSet
 
+class PopupController(app.controller.Controller):
+  """
+  A controller for pop up boxes to notify the user.
+  """
+  def __init__(self, view):
+    app.controller.Controller.__init__(self, view, 'popup')
+    self.view = view
+    self.callerSemaphore = None
+    def noOp(ch, meta):
+      app.log.info('noOp in PopupController')
+    self.commandDefault = noOp
+    self.commandSet = {
+      KEY_ESCAPE: self.changeToMainWindow,
+    }
+
+  def changeToMainWindow(self):
+    self.view.hide()
+    self.host.changeFocusTo(self.host.host)
+    if self.callerSemaphore:
+      self.callerSemaphore.release()
+      self.callerSemaphore = None
```

Comment:
Will enough popup instance use a callerSemaphore to make this part of the base class?

Reply:
No I would guess that only other threads would be using this semaphore. I just thought I would provide the infrastructure so that all the windows can reference the InputWindow's popup window. For the windows that do not need the semaphore, they can simply ignore it.